Makefile

```diff
@@ -26,9 +26,8 @@ CFLAGS += -Wreturn-type
 CFLAGS += -Wwrite-strings
 CFLAGS += -Wcast-align
 
-ifdef GPL
-    CFLAGS += -DUSE_GPL
-endif
+CFLAGS += -DUSE_GPL
+CFLAGS += -fno-toplevel-reorder
```

Comment:
Why?

Reply:
Funny, I just had a conversation with @jwrdegoede about this. In contrast to Doom, which has all its possible game states enumerated  and stored in a single huge `info.c:states[]` array, the Wolf3d engine has them as global variables in `wl_act2.cpp`. The savegame code relies on the order of these variables remaining constant in the executable and the difference of these variables' addresses relative to a certain reference as well. This should be a valid assumption for a given executable, but might change when the code is compiled again - turning the savegames saved wit the prior compilate useless. This compiler flag prevents the reordering of the global variables and thus keeps savegames functional across engine rebuilds.

Reply:
Put that comment in the Makefile. Also file an issue about fixing that, it's really not a safe assumption even with that flag.

Reply:
https://github.com/fabiangreffrath/wolf4sdl/issues/15